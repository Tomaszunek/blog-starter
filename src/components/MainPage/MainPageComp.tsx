import * as React from 'react';
import './mainpage.scss'
import FullScreenCont from '../template/fullScreenCont/fullSreenCont';
import image from 'src/logo.svg'


export default class MainPageComp extends React.Component {    
  public render() {
    return (
      <main>
        <FullScreenCont>          
          <img src={image} alt=""/>
          <img src="./images/logos/greenBlack.png" alt=""/>
        </FullScreenCont>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
        Next things <br/>
      </main>
    );
  }
}


