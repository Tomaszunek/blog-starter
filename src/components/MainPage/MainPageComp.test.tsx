import * as React from 'react';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';
import MainPageComp from './MainPageComp'

describe('Footer Page', () => {
  describe('Component should render right', () => {
    let app:any;             
    beforeEach(() => {
      app = mount(<MainPageComp/>);
    })
    it('render correctly snapshot component', () => {
      const tree = renderer
      .create(<MainPageComp/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
    })    
    it('should contain main containar', () => {
      expect(app.find('main').length).toEqual(1);      
    })
    it('should contain elements', () => {
      expect(app.find('.fullScreenCont').length).toEqual(2);
      expect(app.find('.sliderPhoto').length).toEqual(1);
      expect(app.find('.articleConteiner').length).toEqual(1);
      expect(app.find('.grid.grid-four').length).toEqual(2);      
    })
  });
}); 