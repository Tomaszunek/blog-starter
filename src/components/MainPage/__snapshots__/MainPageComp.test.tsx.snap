// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Page Component should render right render correctly snapshot component 1`] = `
<main>
  <div
    className="fullScreenCont"
  >
    <div
      className="sliderPhoto"
    />
  </div>
  <div
    className="fullScreenCont"
  >
    <div
      className="articleConteiner"
    >
      <div
        className="grid grid-four"
      />
      <div
        className="grid grid-four"
      />
    </div>
  </div>
</main>
`;
